python/pylibcudf/tests/io/test_experimental_hybrid_scan.py

```diff
@@ -0,0 +1,544 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION.
+# SPDX-License-Identifier: Apache-2.0
+import io
+
+import pyarrow as pa
+import pyarrow.parquet as pq
+import pytest
+
+from rmm import DeviceBuffer
+from rmm.pylibrmm.stream import DEFAULT_STREAM, Stream
+
+import pylibcudf as plc
+from pylibcudf.expressions import (
+    ASTOperator,
+    ColumnNameReference,
+    Literal,
+    Operation,
+)
+from pylibcudf.io.experimental import HybridScanReader, UseDataPageMask
+
+
+@pytest.fixture(scope="module")
+def num_rows():
+    """Number of rows in the test table."""
+    return 1000
+
+
+@pytest.fixture(scope="module")
+def row_group_size():
+    """Row group size for parquet files."""
+    return 250
+
+
+@pytest.fixture(scope="module")
+def num_row_groups(num_rows, row_group_size):
+    """Number of row groups in the test parquet file."""
+    return num_rows // row_group_size
+
+
+@pytest.fixture(scope="module")
+def simple_parquet_table(num_rows):
+    """Create a simple PyArrow table for testing."""
+    data = {
+        "col0": pa.array(list(range(num_rows)), type=pa.uint32()),
+        "col1": [f"str_{i}" for i in range(num_rows)],
+        "col2": [float(i) * 1.5 for i in range(num_rows)],
+    }
+    return pa.table(data)
+
+
+@pytest.fixture(scope="module")
+def simple_parquet_bytes(simple_parquet_table, row_group_size):
+    """Create parquet bytes from the simple table."""
+    buf = io.BytesIO()
+    pq.write_table(
+        simple_parquet_table,
+        buf,
+        row_group_size=row_group_size,
+        compression="SNAPPY",
+        use_dictionary=True,
+        write_statistics=True,
```

Comment:
please also use `write_page_index=True` when writing parquet files to be used with the hybrid scan reader. Almost all new filters require page index to be present.